src/main/client/HttpClient.ts

```diff
@@ -0,0 +1,59 @@
+import axios, {AxiosRequestConfig} from 'axios';
+import {logger} from '../wrapper/loggerConfig';
+import * as fs from 'fs';
+import {finished} from 'stream/promises';
+import {Client} from "./Client";
+
+export class HttpClient implements Client {
+    private readonly axiosConfig: AxiosRequestConfig;
+
+    constructor() {
+        this.axiosConfig = {
+            responseType: 'stream',
+            proxy: this.getProxyConfig(),
+        };
+    }
+
+    public getProxyConfig() {
```

Comment:
maybe better to save this value to avoid parsing the URL every time it’s called

Reply:
it's being call once every httpClient object creation